tools/src/main/java/org/apache/kafka/tools/ProducerPerformance.java

```diff
@@ -132,12 +138,16 @@ public static void main(String[] args) throws Exception {
                     transactionStartTime = System.currentTimeMillis();
                 }
 
-
                 if (payloadFilePath != null) {
                     payload = payloadByteList.get(random.nextInt(payloadByteList.size()));
                 }
                 record = new ProducerRecord<>(topicName, payload);
-
+                Enumeration e = headerProps.propertyNames();
+                while (e.hasMoreElements()) {
+                    String key = (String) e.nextElement();
+                    System.out.println(key + " -- " + headerProps.getProperty(key));
+                    record.headers().add(key, headerProps.getProperty(key).getBytes("UTF-8"));
```

Comment:
Instead of adding headers each time, maybe we can pre-create the headers list and pass to ProducerRecord() constructor.

Reply:
Nice catch!!..changes incorporated.